src/webdav_fs.rs

```diff
@@ -0,0 +1,72 @@
+use dav_server::{davpath::DavPath, fs::*, localfs::LocalFs};
+use futures::{future::ready, StreamExt, TryFutureExt};
+use std::path::{Component, Path};
+
+#[derive(Clone)]
+pub struct RestrictedFs {
+    local: Box<LocalFs>,
+    allow_hidden: bool,
+}
```

Comment:
Give this a proper description of what this is supposed to do and why we need it.